packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -0,0 +1,253 @@
+import type { CaipChainId, AccountId } from '@agoric/orchestration';
+import type { Zone } from '@agoric/zone';
+import { type VowTools, VowShape } from '@agoric/vow';
+import type { ZCF, ZCFSeat } from '@agoric/zoe';
+import { M } from '@endo/patterns';
+import type {
+  CCTPTransactionEntry,
+  CCTPTransactionKey,
+  CCTPResolverServiceFacet,
+  CCTPSettlementOfferArgs,
+} from './types.js';
+import { CCTPTransactionDetailsShape } from './types.js';
+
+const trace = makeTracer('Resolver');
+
+/**
+ * Prepare the CCTP Transaction Resolver.
+ *
+ * This is an orchestration component that can be used independently
+ * of portfolio logic to handle CCTP transaction confirmations.
+ *
+ * @param zone - Durable storage zone
+ * @param vowTools - Vow tools for creating promises
+ * @returns A function to create resolver instances with separate facets
+ */
+export const prepareCCTPResolver = (zone: Zone, vowTools: VowTools) => {
+  const resolverZone = zone.subZone('CCTPResolver');
+  const cctpTransactionRegistry = resolverZone.mapStore<
+    CCTPTransactionKey,
+    CCTPTransactionEntry
+  >('cctpTransactionRegistry');
+
+  const ClientFacetI = M.interface('CCTPResolverClient', {
+    registerCCTPTransaction: M.call(M.string(), M.bigint()).returns(VowShape),
+  });
+
+  const ServiceFacetI = M.interface('CCTPResolverService', {
+    settleCCTPTransaction: M.call(
+      M.string(),
+      M.string(),
+      M.bigint(),
+      M.string(),
+    ).returns(),
+  });
+
+  const DebugFacetI = M.interface('CCTPResolverDebug', {
+    getPendingTransactions: M.call().returns(M.arrayOf(M.string())),
+  });
+
+  return resolverZone.exoClassKit(
+    'CCTPResolver',
+    {
+      client: ClientFacetI,
+      service: ServiceFacetI,
+      debug: DebugFacetI,
+    },
+    () => ({}),
+    {
+      client: {
+        registerCCTPTransaction(destinationAddress: AccountId, amount: bigint) {
+          const key = `${destinationAddress}:${amount}` as CCTPTransactionKey;
+
+          if (cctpTransactionRegistry.has(key)) {
+            trace(`CCTP transaction already registered: ${key}`);
+            return cctpTransactionRegistry.get(key).vowKit.vow;
+          }
+
+          const vowKit = vowTools.makeVowKit<void>();
+
+          cctpTransactionRegistry.init(
+            key,
+            harden({
+              destinationAddress,
+              amountValue: amount,
+              vowKit,
+            }),
+          );
+
+          trace(`Registered pending CCTP transaction: ${key}`);
+          return vowKit.vow;
+        },
+      },
+      service: {
+        settleCCTPTransaction(
+          chainId: CaipChainId,
+          remoteAddress: `0x${string}`,
+          amountValue: bigint,
+          status: 'confirmed' | 'failed' | 'pending',
+        ) {
+          const destinationAddress: AccountId = `${chainId}:${remoteAddress}`;
+          const key =
+            `${destinationAddress}:${amountValue}` as CCTPTransactionKey;
```

Comment:
declaring `destinationAddress` separately makes me wonder what it's used for. "find all references" shows: nothing.

```suggestion
          const key =
            `${chainId}:${remoteAddress}:${amountValue}` as CCTPTransactionKey;
```